### PR TITLE
(fix) requiredPrivilege should not display an error if privilege is undefined

### PR DESCRIPTION
**Pull request by ibacher**
## Requirements
- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.

#### For changes to apps
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

Basically, this removed an error message I accidentally introduced. If `userHasPrivilege()` is called and the `requiredPrivilege` is `undefined`, then `userHasPrivilege()` should simply be a no-op returning `true`.

### CODE REVIEW

File: packages/framework/esm-api/src/shared-api-objects/current-user.ts [hunk 1]
```diff
@@ -134,6 +134,7 @@ function userHasPrivilege(
     return requiredPrivilege.every(
       (rp) => !isUndefined(user.privileges.find((p) => rp === p.display))
     );
+  } else if (isUndefined(requiredPrivilege)) {
   } else {
```
Comment (vasharma05) [inline packages/framework/esm-api/src/shared-api-objects/current-user.ts:138]:
```suggestion
  } else if (!isUndefined(requiredPrivilege)) {
  }
```

**Review by vasharma05 — APPROVED** [review]

File: packages/framework/esm-api/src/shared-api-objects/current-user.ts (hunk 1 above)
Comment (vasharma05) [reply packages/framework/esm-api/src/shared-api-objects/current-user.ts:138]: An empty if block seems abrupt.

Comment (ibacher) [reply packages/framework/esm-api/src/shared-api-objects/current-user.ts:138]: It's basically to stop being processed by the next block, which logs an error...

Comment (ibacher) [inline packages/framework/esm-api/src/shared-api-objects/current-user.ts:138]:
```suggestion
  } else if (!isUndefined(requiredPrivilege)) {
```
